sprokit/processes/core/motion_detector_process.cxx

```diff
@@ -0,0 +1,181 @@
+#include <sprokit/pipeline/process_exception.h>
+
+namespace kwiver {
+
+create_config_trait( algo, std::string, "", "Algorithm configuration subblock" );
+
+//----------------------------------------------------------------
+// Private implementation class
+class motion_detector_process::priv
+{
+public:
+  priv();
+  ~priv();
+
+  vital::algo::motion_detector_sptr m_algo;
+  kwiver::vital::wall_timer m_timer;
+
+}; // end priv class
+
+
+// ==================================================================
+motion_detector_process::
+motion_detector_process( kwiver::vital::config_block_sptr const& config )
+  : process( config ),
+    d( new motion_detector_process::priv )
+{
+  make_ports();
+  make_config();
+}
+
+
+motion_detector_process::
+~motion_detector_process()
+{
+}
+
+
+// ------------------------------------------------------------------
+void
+motion_detector_process::
+_configure()
+{
+  scoped_configure_instrumentation();
+
+  vital::config_block_sptr algo_config = get_config();
+
+  // Check config so it will give run-time diagnostic of config problems
+  if ( ! vital::algo::motion_detector::check_nested_algo_configuration_using_trait( algo, algo_config ) )
+  {
+    throw sprokit::invalid_configuration_exception( name(), "Configuration check failed." );
+  }
+
+  vital::algo::motion_detector::set_nested_algo_configuration_using_trait( algo, algo_config, d->m_algo );
+
+  if ( ! d->m_algo )
+  {
+    throw sprokit::invalid_configuration_exception( name(), "Unable to create motion detector algorithm" );
+  }
+}
+
+
+// ------------------------------------------------------------------
+void
+motion_detector_process::
+_step()
+{
+  d->m_timer.start();
+
+  //TODO, handle case where this is optionally provided
+  //auto ts = grab_from_port_using_trait( timestamp );
+
+  auto input = grab_from_port_using_trait( image );
+  kwiver::vital::image_container_sptr result;
+
+  {
+    scoped_step_instrumentation();
+
+    kwiver::vital::timestamp ts;
+    if (has_input_port_edge_using_trait( timestamp ) )
+    {
+      ts = grab_from_port_using_trait( timestamp );
+    }
+
+    bool reset(false);
+    if (has_input_port_edge_using_trait( coordinate_system_updated ) )
+    {
+      reset = grab_from_port_using_trait( coordinate_system_updated );
+    }
+
+    result = d->m_algo->process_image( ts, input, reset );
+  }
+
+  push_to_port_using_trait(motion_heat_map , result );
+
+  d->m_timer.stop();
+  double elapsed_time = d->m_timer.elapsed();
+  LOG_DEBUG( logger(), "Total processing time: " << elapsed_time << " seconds");
+}
+
+
+// ------------------------------------------------------------------
+void
+motion_detector_process::
+make_ports()
+{
+  // Set up for required ports
+  sprokit::process::port_flags_t required;
+  sprokit::process::port_flags_t optional;
+  sprokit::process::port_flags_t opt_static;
+
+  required.insert( flag_required );
+  opt_static.insert( flag_input_static );
```

Comment:
Doesn't look like `opt_static` is used

Reply:
Yup, this can be removed here (probably pasted from a reference source).